### PR TITLE
Sha3 operation

### PR DESCRIPTION
**Pull request by natdurlik**
Implemented sha3 operation.

### CODE REVIEW

File: tests/transpilation/sha3.opcode [hunk 1]
```diff
@@ -0,0 +1,3 @@
+PUSH1 0x20
+PUSH1 0x0
+SHA3
```
Comment (murcake) [inline tests/transpilation/sha3.opcode:3]: Is it true that you compute keccak of 0 here?

File: src/Operations/SHA3.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+                let (local msize) = update_msize(msize, {offset}, {length})
+                let (local memval) = create_from_memory({offset}, {length})
+                local high
+                local low
+                %{{
+                    from Crypto.Hash import keccak
+
+                    keccak_hash = keccak.new(digest_bits=256)
+                    arr = []
+                    arr_length = {length}//16 if {length} % 16 == 0 else {length}//16+1
```
Comment (murcake) [inline src/Operations/SHA3.py:24]: Just a note: I don't know if you consider it more readable, but ceil division can be expressed as `({length} - 1) // 16 + 1`

**Review by murcake — COMMENTED** [review]
LGTM. I would suggest though to squash all commits into one, since they basically fix each other

File: tests/transpilation/sha3.opcode (hunk 1 above)
Comment (natdurlik) [reply tests/transpilation/sha3.opcode:3]: Yes

**Pull request merged**